src/backend/dx12/src/command.rs

```diff
@@ -283,6 +283,11 @@ pub struct CommandBuffer {
 
     // Re-using allocation for the image-buffer copies.
     copies: Vec<Copy>,
+
+    // D3D12 only allows setting all viewports or all scissors at once, not partial updates.
+    // So we must cache the implied state for these partial updates.
+    viewport_cache: SmallVec<[d3d12::D3D12_VIEWPORT; 16]>,
```

Comment:
Please use `D3D12_VIEWPORT_AND_SCISSORRECT_OBJECT_COUNT_PER_PIPELINE` instead of 16